src/preview.cpp

```diff
@@ -1012,7 +1012,7 @@ namespace {
 
     DataBuf makePnm(uint32_t width, uint32_t height, const DataBuf &rgb)
     {
-        const long expectedSize = static_cast<long>(width * height * 3);
+        const long expectedSize = static_cast<long>(width) * static_cast<long>(height) * 3;
```

Comment:
Should also work like this, right?
```cpp
const long expectedSize = static_cast<long>(width) * height * 3;
```
But still no remedy against the overflow (on Win, where `size(long)` = `size(uint32_t)`).

Reply:
Oh right. I will change it. Thanks.

Reply:
Hm, did you try if
```cpp
const long expectedSize = static_cast<long>(width) * height * 3;
```
removes the warning. Or, if not, if
```
const long expectedSize = (long)width * height * 3;
```
does?

Reply:
To be honest, I don't think this should be changed in this way. 

Consider currently the current outcome if we get an overflow: `width * height * 3` is larger then `std::numeric_limits<uint32_t>::max()` => the overflow is well defined (everything is unsigned). We'll get UB on Windows since `sizeof(long) == sizeof(uint32_t)` but everywhere else we won't. 
Now with this change: `width` is treated as a `long` thus the multiplication can trigger UB on any platform, as we are multiplying signed integers.

Don't get me wrong: this code is currently doing things the wrong way (sizes should really not be `long` but `size_t`) and should be fixed (unfortunately that involves replacing `DataBuf`). But this fix trades UB on Windows for UB on every platform. Also, if there is an actual overflow, the code will abort in the following if.

Reply:
I think you analysis is right. Here I was just playing around with PVS-Studio and I did not think much about the changes. 

Since any of the issues is critical, I'll close the PR.